src/APIController.react.js

```diff
@@ -97,14 +95,26 @@ class UnconnectedContainer extends Component {
                 </div>
             );
         } else if (appLifecycle === getAppState('HYDRATED')) {
+            window.console.log('hydrated return');
             return (
                 <div id="_dash-app-content">
-                    <TreeContainer layout={layout} />
+                    <TreeContainer layout={layout} loading={false} />
                 </div>
             );
         }
-
-        return <div className="_dash-loading">{'Loading...'}</div>;
+        if (isEmpty(layout)) {
+            return (
+                <div id="_dash-app-content">
+                    <TreeContainer layout={layout} loading={true} />
+                </div>
+            );
+        }
+        window.console.log('return at end of render()');
```

Comment:
:zap:

Reply:
Yeah, this isn't ready for review yet, still WIP!

Reply:
I was just browsing to see how it would go with #92 and #91 and noticed that.